sonic-net/sonic-mgmt
# [VM]: add new topo wan-4link (4 link between each device based on wan-pub)

### Description of PR
In order to support load balancing between devices, add a new topo based on wan-pub that each device is connected 4 physical links in one port channel.


Summary:
Fixes # (issue)

### Type of change





### Back port request

### Approach
#### What is the motivation for this PR?
Add a new topo based on wan-pub that support multi-links between devices.
#### How did you do it?

1. Add a new topo (wan-4link)
2. Since wan topo is not so symmetric as DC, move wan related configuration which would not be covered in other parts under `wan_dut_configuration`.

#### How did you verify/test it?

1. Start VM using command `./testbed-cli.sh -m veos_vtb -k vcisco -t vtestbed.yaml start-topo-vms vms-kvm-wan-4link password.txt `
2. Add topo using command `./testbed-cli.sh -t vtestbed.yaml -m veos_vtb -k vcisco add-topo vms-kvm-wan-4link password.txt -e eos_batch_size=1`
3. Deploy DUT minigraph using command  `./testbed-cli.sh -t vtestbed.yaml -m veos_vtb deploy-mg vms-kvm-wan-4link veos_vtb password.txt`
4. Check lldp neighbor in DUT, each neighbor has 4 ports advertised to this DUT.
5. Setup wan-pub topo to check port channel would be setup successfully.

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?
`wan-pub`, `wan-4link`
### Documentation
